app/controllers/api/migrations.php

```diff
@@ -0,0 +1,824 @@
+    ->groups(['api', 'migrations'])
+    ->label('scope', 'public')
+    ->label('sdk.namespace', 'migrations')
+    ->label('sdk.method', 'listFirebaseProjects')
+    ->label('sdk.description', '')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_MIGRATION_FIREBASE_PROJECT_LIST)
+    ->inject('user')
+    ->inject('response')
+    ->inject('project')
+    ->inject('dbForConsole')
+    ->inject('request')
+    ->action(function (Document $user, Response $response, Document $project, Database $dbForConsole, Request $request) {
+        $firebase = new OAuth2Firebase(
+            App::getEnv('_APP_MIGRATIONS_FIREBASE_CLIENT_ID', ''), 
+            App::getEnv('_APP_MIGRATIONS_FIREBASE_CLIENT_SECRET', ''), 
+            $request->getProtocol() . '://' . $request->getHostname() . "/v1/migrations/firebase/redirect"
+        );
+
+        $accessToken = $user->getAttribute('migrationsFirebaseAccessToken');
+        $refreshToken = $user->getAttribute('migrationsFirebaseRefreshToken');
+        $accessTokenExpiry = $user->getAttribute('migrationsFirebaseAccessTokenExpiry');
+
+        $isExpired = new \DateTime($accessTokenExpiry) < new \DateTime('now');
+        if ($isExpired) {
+            $firebase->refreshTokens($refreshToken);
+
+            $accessToken = $firebase->getAccessToken('');
+            $refreshToken = $firebase->getRefreshToken('');
+
+            $verificationId = $firebase->getUserID($accessToken);
+
+            if (empty($verificationId)) {
+                throw new Exception(Exception::GENERAL_RATE_LIMIT_EXCEEDED, "Another request is currently refreshing OAuth token. Please try again.");
+            }
+
+            $user = $user
+                ->setAttribute('migrationsFirebaseAccessToken', $accessToken)
+                ->setAttribute('migrationsFirebaseRefreshToken', $refreshToken)
+                ->setAttribute('migrationsFirebaseAccessTokenExpiry', DateTime::addSeconds(new \DateTime(), (int)$firebase->getAccessTokenExpiry('')));
+
+            $dbForConsole->updateDocument('users', $user->getId(), $user);
+        }
+
+        $projects = $firebase->getProjects($accessToken);
+
+        $output = [];
+        foreach ($projects as $project) {
+            $output[] = [
+                'displayName' => $project['displayName'],
+                'projectId' => $project['projectId'],
+            ];
+        }
+
+        $response->dynamic(new Document([
+            'projects' => $output,
+            'total' => count($output),
+        ]), Response::MODEL_MIGRATION_FIREBASE_PROJECT_LIST);
+    });
+
+App::get('/v1/migrations/firebase/deauthorize')
+    ->desc('Revoke Appwrite\'s authorization to access Firebase Projects')
+    ->groups(['api', 'migrations'])
+    ->label('scope', 'public')
+    ->label('sdk.namespace', 'migrations')
+    ->label('sdk.method', 'firebaseDeauthorize')
```

Comment:
Let's change this to `deleteFirebaseAuth` to match the `createFirebaseAuth` method